Test/QuickCheck/Modifiers.hs

```diff
@@ -343,4 +355,100 @@ instance (Arbitrary a, ShrinkState s a) => Arbitrary (Shrinking s a) where
 #endif /* NO_MULTI_PARAM_TYPE_CLASSES */
 
 --------------------------------------------------------------------------
+-- | @ASCII@.
+newtype ASCII = ASCII {getASCII :: Char}
+  deriving ( Eq, Ord, Show, Read )
+
+asciiGen :: Gen Char
+asciiGen = choose ('\0', '\127')
+
+instance Arbitrary ASCII where
+  arbitrary = ASCII `fmap` asciiGen
+
+--------------------------------------------------------------------------
+-- | @Latin1@: latin1 Char, same Arbitrary as Char itself.
+newtype Latin1 = Latin1 {getLatin1 :: Char}
```

Comment:
Yeah, I'm also :+1: for switching to Unicode right away. If this will cause breakage it's because people have latent Unicode bugs in their codebase for which they're not aware, and getting tests to fail will be useful.

I added the Latin1 newtype only so that people would have an easy migration path if they want to retain the existing logic and don't want to fix their code

I don't think it adds much value over the ASCII newtype, but you never know on which behavior downstream code might be relying upon

btw, one small difference is the shrink: I think that the set of unicode codepoints is too big to meaningfully explore the space when shrinking. The same might not be true for the ASCII newtype, but for simplicity I didn't bother replicating Latin1's logic into it